client.go

```diff
@@ -526,7 +537,16 @@ func (c *Client) send(r *Request) (*amqp.Delivery, error) {
 	timeoutChan := r.startTimeout()
 
 	c.debugLog("client: queuing request %s", r.Publishing.CorrelationId)
-	c.requests <- r
+
+	select {
```

Comment:
Yeah, I thought about using a go routine but it felt more intuitive to have timeout functionality.

Reply:
Yea, every message is blocking other messages and they are sent sequentially. This is because publishing on a amqp channel isn't safe for concurrent use. I did some benchmarking on this and having more workers doesn't really increase the performance.

Using a goroutine around the `c.request <- r` would just cause the send function to block a bit later, and if sending a lot of requests using the client create a lot of goroutines.